tkg/client/common.go

```diff
@@ -323,9 +322,21 @@ func (c *TkgClient) SetAllowLegacyClusterConfiguration() string {
 		log.V(6).Infof("Setting %v to %q", constants.ConfigVariableAllowLegacyCluster, allowLegacyCluster)
 		c.TKGConfigReaderWriter().Set(constants.ConfigVariableAllowLegacyCluster, allowLegacyCluster)
 	} else {
-		log.V(6).Infof("Info: %v configuration already set to %q", constants.ConfigVariableAllowLegacyCluster, value)
 		// ALLOW_LEGACY_CLUSTER is explicitly set in cluster config file
-		allowLegacyCluster = value
+		log.V(6).Infof("Info: %v configuration already set to %q", constants.ConfigVariableAllowLegacyCluster, value)
+		if value != "true" && value != "false" {
+			log.Warningf("Seem like you have set %v, but it's not a valid value. It should be true or false", constants.ConfigVariableAllowLegacyCluster)
+			// Set ALLOW_LEGACY_CLUSTER to a valid value
+			if !c.IsFeatureActivated(constants.FeatureFlagAllowLegacyCluster) {
```

Comment:
I am confused why there are feature flag allow-legacy-cluster and variable ALLOW_LEGACY_CLUSTER?

Reply:
@DanielXiao , according to the [design doc](https://docs.google.com/document/d/1VV_3z1d2OmJC9CoXMszYOvS1cFMziUL1nupMlIJbRM8/edit#), if user wants to create a legacy based cluster, then either use `tanzu config set features.cluster.allow_legacy_cluster true` command, or add variable `ALLOW_LEGACY_CLUSTER: "true"` into config file to active legacy mode.